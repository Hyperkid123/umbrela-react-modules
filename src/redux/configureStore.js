import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
