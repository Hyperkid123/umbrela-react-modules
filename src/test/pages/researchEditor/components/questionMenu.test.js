import React from 'react'
import QuestionMenu, {QuestionMenu as Snapshot} from '../../../../pages/researchEditorModule/components/questionMenu';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('Question Menu component', () => {
  const initialState  = {
    editor: {
      activeSheet: {
        sheetId: 10
      }
    },
    ui: {
      draggingElement: false,
    },
    questions: {
      activeQuestion: {
        questionId: 10
      }
    },
    locale: {
      languages: [{
        name: "English",
        code: 'en',
        active: true,
      }],
      translations: {}
    }
  };

  const mockStore = configureStore();
  it('should create component correctly', () => {
    let store = mockStore(initialState);
    let wrapper = shallow(<QuestionMenu store={store}/>);
    expect(wrapper).toBeDefined();
  });

  it('should render correctly empty', () => {
    const tree = shallow(
      <Snapshot
        activeSheet={{
          title: 'foo'
        }}
        getQuestions={jest.fn()}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should render correctly with questions', () => {
    const tree = shallow(
      <Snapshot
        questions={[{
          questionId: 1
        }, {
          questionId: 2
        }]}
        getQuestions={jest.fn()}
        translate={jest.fn()}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should get new questions on sheet change', () => {
    const getQuestions = jest.fn();
    const wrapper = shallow(
      <Snapshot
        activeSheet={{
          title: 'foo'
        }}
        activeSheetId={0}
        getQuestions={getQuestions}
      />
    );
    expect(getQuestions.mock.calls.length).toEqual(1);
    wrapper.setProps({activeSheetId: 1});
    expect(getQuestions.mock.calls.length).toEqual(2);
  })
});
