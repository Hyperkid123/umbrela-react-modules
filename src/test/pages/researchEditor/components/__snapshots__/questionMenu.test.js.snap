// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Menu component should render correctly empty 1`] = `
<h2
  className="sc-csuQGl hmEPUJ"
>
  Loading
</h2>
`;

exports[`Question Menu component should render correctly with questions 1`] = `
<section
  className="sc-bxivhb bDsTwg"
>
  <styled.div
    column={true}
    grow={true}
  >
    <styled.div
      fullWidth={true}
    >
      <styled.h2 />
      <styled.ul>
        <DragSource(DropTarget(SheetDraggableCard))
          index={0}
          key="1"
          moveCard={[Function]}
        >
          <MenuListItem
            active={false}
            key="1"
            sheetId={1}
          />
        </DragSource(DropTarget(SheetDraggableCard))>
        <DragSource(DropTarget(SheetDraggableCard))
          index={1}
          key="2"
          moveCard={[Function]}
        >
          <MenuListItem
            active={false}
            key="2"
            sheetId={2}
          />
        </DragSource(DropTarget(SheetDraggableCard))>
      </styled.ul>
    </styled.div>
  </styled.div>
</section>
`;
